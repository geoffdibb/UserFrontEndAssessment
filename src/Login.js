import React, { Component } from 'react';
import axios from 'axios';


export default class Login extends Component {

    constructor() {
        super();
        this.state = {
            Username: "",
            Password: "",
            data: ""
        }
    }


    getone = (e) => {
        e.preventDefault();
        let newItem = {
            username: e.target[0].value,
            password: e.target[1].value
        }


        axios
            .get("http://localhost:5000/User/name/" + newItem.username + "/" + newItem.password)
            .then(response => {
                this.setState({
                    data: response.data
                });
            }).catch(err => console.log(err));
    };


    render() {
        return (
            <div>
                <form onSubmit={this.getone}>

                    <br></br>
                    <label for="Username">Username: </label>
                    <input type="text" id="Username" name="Username" required />
                    <br></br>

                    <label for="Password">Password: </label>
                    <input type="Password" id="Password" name="Password" required />
                    <br></br>

                    <input type="submit" value="Login" />
                    <p id="systemResponse"> </p>
                </form>

                {this.state.data.Status}

            </div>
        )

    }
}