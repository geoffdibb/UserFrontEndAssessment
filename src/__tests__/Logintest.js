import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Login from '../Login';
import renderer from 'react-test-renderer';


it('snapshot Login', () => {
  const tree = renderer.create(<Login/>).toJSON()
  expect(tree).toMatchSnapshot();
})
