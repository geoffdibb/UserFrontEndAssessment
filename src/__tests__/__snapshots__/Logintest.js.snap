// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Login 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <br />
    <label
      for="Username"
    >
      Username: 
    </label>
    <input
      id="Username"
      name="Username"
      required={true}
      type="text"
    />
    <br />
    <label
      for="Password"
    >
      Password: 
    </label>
    <input
      id="Password"
      name="Password"
      required={true}
      type="Password"
    />
    <br />
    <input
      type="submit"
      value="Login"
    />
    <p
      id="systemResponse"
    >
       
    </p>
  </form>
</div>
`;
